### ISSUE TITLE
Make airblast drill consume cryofluid for boost

### ISSUE THREAD

**Issue by Slava0135** [Anuken/Mindustry-Suggestions#2141]
**Describe what you would like changed, and why.**
It's not a duplicate of #156 because I want to keep laser drill as it is. So the problem is - laser drill is only being used as a way to get thorium to replace them right after with airblast one.

**Describe the changes you want to propose. Include possible alternatives.**
So idea is to make airblast drill act a little bit differently from other drills by adding cryo requirement. It will give laser drill more attention.

**Comment by itcannotbe** [comment]
Or you could just make water not an option for airblast.

**Comment by Slava0135** [comment]
> Or you could just make water not an option for airblast.

Lower boost value?

**Comment by itcannotbe** [comment]
Probably increase water consumption and make it a requirement instead of an option to reduce it's flexibility.

**Comment by itcannotbe** [comment]
And yes, only an idiot would use the airblast without water.

**Comment by Slava0135** [comment]
> And yes, only an idiot would use the airblast without water.

And I don't see this happening so yes

**Comment by Qvertyes** [comment]
Titanium one of most consumed resources. If change this issue by this way, i think we need 3x3 cryofluid mixer with more output of cryofluid or something

**Comment by Slava0135** [comment]
> Titanium one of most consumed resources. If change this issue by this way, i think we need 3x3 cryofluid mixer with more output of cryofluid or something

1 cryofluid mixer can feed 2 air blast drills so I don't see any issues

**Comment by Sense101** [comment]
I really like this idea it makes a lot of sense

**Comment by Coolhandluke348** [comment]
No... why... are... you... making... every... thing... in... this... game... more... tedious

**Comment by Coolhandluke348** [comment]
An alternate solution, instead of nerfing water, have a significant boost from using cryo over water

**Comment by BlueWolf3682** [comment]
🚯

**Comment by stuffyAI** [comment]
This makes no sense from a balancing standpoint?

The airblast drill is simply 4 by 4, so 1 bigger to each side compared to the laser drill.

BUT the rate of extraction per cell remains the same, e.g if i have a 4x4 patch of ore **Laser and Airblast drills will give you equal amounts of ore**, you obviously only need to use 4 laser drills here to fill it out.

**Comment by Slava0135** [comment]
> This makes no sense from a balancing standpoint?
> 
> The airblast drill is simply 4 by 4, so 1 bigger to each side compared to the laser drill.
> 
> BUT the **rate of extraction per cell remains the same**, e.g if i have a 4x4 patch of ore **Laser and Airblast drills will give you equal amounts of ore**, you obviously only need to use 4 laser drills here to fill it out.

Actuaaly not. Airblast has a bigger boost from water making laser drill useless unless there is no space AB

**Comment by stuffyAI** [comment]
> Airblast has a bigger boost from water making laser drill useless

Well that's the literal point isn't it?
You upgrade your drill by spending more and better ressources whilst allocating more power to water and mining production?
I think the decision to provide 26% more mining rate for around 63% more power cost (plus whatever you put into producing the additional water) whilst also making airblast harder to user due to size constrains is both balanced and sensible.

**Comment by Sense101** [comment]
> > Airblast has a bigger boost from water making laser drill useless
> 
> Well that's the literal point isn't it?
> You upgrade your drill by spending more and better ressources whilst allocating more power to water and mining production?
> I think the decision to provide 26% more mining rate for around 63% more power cost (plus whatever you put into producing the additional water) whilst also making airblast harder to user due to size constrains is both balanced and sensible.

the thing is, it isn't more power because you only use one airblast where you use three laser drills or more.
Its basically just faster drilling for nothing

**Comment by stuffyAI** [comment]
> because you only use one airblast where you use three laser drills or more.

It is power?
Do the math, airblast consumes 35% more power than a laser drill per cell, assuming that we have the perfect scenario where the airblast actually occupies all cells efficiently where ores are, it also uses more water per cell so you have to get that, probablty by spending more power on water extractors or pumps.

**Comment by Bohaska** [comment]
> An alternate solution, instead of nerfing water, have a significant boost from using cryo over water

hmm....

**Comment by Coolhandluke348** [comment]
> hmm.... the code isn't made to easily allow me to give different boosts from using different liquids. Perhaps increase the liquid multiplier by even more, but only allow it to accept cryofluid.

Turrets already have that system, it wouldn’t be that hard to implement, and having to input titanium to every drill you want to use efficiently seems ridiculous when you already use titanium for so much

**Comment by Bohaska** [comment]
> having to input titanium to every drill you want to use efficiently

only applies to airblast drills....

**Comment by Bohaska** [comment]
>it wouldn’t be that hard to implement

lemme try. I wouldn't add the boosts yet, but i will try to split up the liquid boost code into water boost and cryofluid boost code

**Comment by itcannotbe** [comment]
Anything hard coded is dirty code. Since the game is modable via an api of sorts at this moment.

**Comment by YoMamaLlama1** [comment]
I mean, its not like airblast drills fill up a whole tita line with water boost right?

**Comment by G0nTar** [comment]
I mean, make it a bit more expensive all around to balance it out, Make it so it overheats and explodes if cooling gets shutdown (just like thorium reactor) while it drills, like a small reactor explosion that throws fire around it (or just turns on fire because it's so bi fast and hot), that would make sence and most important be FUN, or Epic, and balance it all together (we can now increase its boost output a bit too, since it's now a step higher on the drill power ladder and only after nerfing it's cost and having a hazard mining thing we could boost anything to compensate) 

That way the cryo fluid would make more sence into a really hot turbine drill that would heat up and explode or burn in flames if not cooled properly, it sounds exiting and epic to have such a dangerous n powerful drill on the player mining equipment, and would make *Laser drill* the "safe" approach
